DFC.Digital/DFC.Digital.Repository.SitefinityCMS/Modules/JobProfilesRelatedCareersRepository.cs

```diff
@@ -51,7 +52,7 @@ public IEnumerable<JobProfileRelatedCareer> GetRelatedCareers(DynamicContent job
             {
                 if (relatedCareerProfilesItem.Any())
                 {
-                    return relatedCareerProfilesItem.Select(item => converter.ConvertFrom(item));
+                    return relatedCareerProfilesItem.ToList().Where(x => !dynamicContentExtensions.GetFieldValue<bool>(x, IsImportedField)).Select(item => converter.ConvertFrom(item));
```

Comment:
Why ToList()? that would return all the data back and then we are filtering in memory, cant we just where clause it?

Reply:
We need to pull the list into memory before we where clause it. The data provider will fail to interpret the filter on the query hence the to list extension